3rdparty/udis86/libudis86/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# UJIT: CMake file for udis86 to avoid dependencies on autoconf and libtool.
+# UJIT: Based on https://github.com/vmt/udis86/pull/99/files by hasherezade.
+# UJIT: Copyright (C) 2020 Anton Soldatov.
+
+set(ITAB_H "${CMAKE_CURRENT_LIST_DIR}/itab.h")
+set(ITAB_C "${CMAKE_CURRENT_LIST_DIR}/itab.c")
+
+if(NOT EXISTS ${ITAB_H} OR NOT EXISTS ${ITAB_C})
+  message(FATAL_ERROR "You must generate files: ${ITAB_H}, ${ITAB_C} before you start.")
+endif()
+
+set(SOURCES_LIBUDIS86
+  decode.c
+  itab.c
+  syn.c
+  syn-att.c
+  syn-intel.c
+  udis86.c
+)
+
+add_library(libudis86 STATIC ${SOURCES_LIBUDIS86})
```

Comment:
Why is it necessary to build this dependency as a static library?
Have it always been build that way for both static and dynamic libujit?

Reply:
Another question: how is this target used? I don't see any other CMake targets depending on it, which means that it should be built manually?

Reply:
> Another question: how is this target used? I don't see any other CMake targets depending on it, which means that it should be built manually?

This target has never been used explicitly since the time we began to depend on `udis86`. Only object files are needed.

Reply:
> Have it always been build that way for both static and dynamic libujit?

Yes, it has since the day the dependency was added. The reason was to use object files directly to avoid dependencies on the `libudis86` (as a static or a dynamic library). So I guess changing this behavior is out of scope of this patch.